mongo/integration/unified/event.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package unified
+
+import (
+	"strings"
+
+	"go.mongodb.org/mongo-driver/event"
+)
+
+type monitoringEventType string
+
+const (
+	commandStartedEvent           monitoringEventType = "CommandStartedEvent"
+	commandSucceededEvent         monitoringEventType = "CommandSucceededEvent"
+	commandFailedEvent            monitoringEventType = "CommandFailedEvent"
+	poolCreatedEvent              monitoringEventType = "PoolCreatedEvent"
+	poolClearedEvent              monitoringEventType = "PoolClearedEvent"
+	poolClosedEvent               monitoringEventType = "PoolClosedEvent"
+	connectionCreatedEvent        monitoringEventType = "ConnectionCreatedEvent"
+	connectionReadyEvent          monitoringEventType = "ConnectionReadyEvent"
+	connectionClosedEvent         monitoringEventType = "ConnectionClosedEvent"
+	connectionCheckOutFailedEvent monitoringEventType = "ConnectionCheckOutFailedEvent"
+	connectionCheckedOutEvent     monitoringEventType = "ConnectionCheckedOutEvent"
+	connectionCheckedInEvent      monitoringEventType = "ConnectionCheckedInEvent"
+)
+
+func monitoringEventTypeFromString(eventStr string) (monitoringEventType, bool) {
```

Comment:
Nit: this is only called twice, but would returning an `error` be preferable to a `bool`?

Reply:
I had this return bool rather than error because it mimics the semantics of a map and other map-like types (e.g `sync.Map`), which all return `(value, ok)` from their `get` functions.